api/fossa/tar.go

```diff
@@ -38,6 +38,16 @@ type SignedURL struct {
 	SignedURL string
 }
 
+// UploadTarballDependency uploads the directory specified to be treated on FOSSA as a dependency.
+func UploadTarballDependency(dir string, upload, rawLicenseScan bool) (Locator, error) {
```

Comment:
it looks like this function is only ever called with `rawLicenseScan` as `true`. Perhaps it shouldn't be a parameter?

Reply:
I was a little split on that decision here. I amended the PR comment to address this. Ideally, we will give users the option to upload projects for the sake of running full backend dependency analysis on it, but that would require adding some hacky flags on `fossa analyze`. To do this right I need to refactor the tar uploader into its own "analyzer".
>This PR enforces that all raw type uploads will only have a license scan run on them. In order to take advantage of the ability to upload files for dependency analysis, I would like to break the tarball uploader out into a separate analyzer to keep configuration under control.